UnityAds/src/main/java/com/mopub/mobileads/UnityBaseAd.java

```diff
@@ -0,0 +1,119 @@
+import static com.mopub.mobileads.UnityAdsConstants.LOG_PLACEMENT_ID_MISSING;
+import static com.mopub.mobileads.UnityAdsConstants.PLACEMENT_ID_KEY;
+import static com.mopub.mobileads.UnityAdsConstants.ZONE_ID_KEY;
+
+public abstract class UnityBaseAd extends BaseAd {
+    private static final String ADAPTER_NAME = UnityBaseAd.class.getSimpleName();
+    protected Activity mActivity;
+    protected String mPlacementId = "";
+
+    protected abstract String getDefaultPlacementId();
+
+    @Nullable
+    @Override
+    protected LifecycleListener getLifecycleListener() {
+        return null;
+    }
+
+    @NonNull
+    @Override
+    protected String getAdNetworkId() {
+        return mPlacementId;
+    }
+
+    @Override
+    protected boolean checkAndInitializeSdk(@NonNull Activity launcherActivity, @NonNull AdData adData) {
+        Preconditions.checkNotNull(launcherActivity);
+        Preconditions.checkNotNull(adData);
+
+        mActivity = launcherActivity;
+        return UnityAdsInitializer.getInstance().checkOrInitUnityAds(mActivity, adData.getExtras());
+    }
+
+    protected void initializeAndFailAdRequest(@NonNull final Context context, @NonNull final AdData adData) {
+        String gameId = adData.getExtras().get(GAME_ID_KEY);
+        if (gameId == null) {
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_GAME_ID_MISSING);
+            return;
+        }
+
+        UnityAdsInitializer.getInstance().initializeUnityAds(context, gameId, new IUnityAdsInitializationListener() {
+            @Override
+            public void onInitializationComplete() {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_INIT_SUCCESS);
+            }
+
+            @Override
+            public void onInitializationFailed(UnityAds.UnityAdsInitializationError unityAdsInitializationError, String s) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_INIT_FAILED_WITH_ERROR + s);
+            }
+        });
+
+        if (mLoadListener != null) {
+            MoPubLog.log(getAdNetworkId(), CUSTOM, ADAPTER_NAME,
+                    LOG_CANNOT_LOAD_BEFORE_INIT);
+            mLoadListener.onAdLoadFailed(MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
+        }
+    }
+
+     public String placementIdForServerExtras(Map<String, String> serverExtras) {
+        String placementId = null;
+        if (serverExtras.containsKey(PLACEMENT_ID_KEY)) {
+            placementId = serverExtras.get(PLACEMENT_ID_KEY);
+        } else if (serverExtras.containsKey(ZONE_ID_KEY)) {
+            placementId = serverExtras.get(ZONE_ID_KEY);
+        }
+        return TextUtils.isEmpty(placementId) ? getDefaultPlacementId() : placementId;
+    }
+
+    @Override
+    protected void load(@NonNull Context context, @NonNull AdData adData) {
+        Preconditions.checkNotNull(context);
+        Preconditions.checkNotNull(adData);
+
+        if (!UnityAds.isInitialized()) {
+            initializeAndFailAdRequest(context, adData);
```

Comment:
I like how you extracted this elsewhere as this is clearly some strange legacy behaviour of us lazy init our SDK and silently fail the first load request. :-) (for future behaviour change unless this is what MoPub documented the intention is of course)